src/webapp/components/survey/hook/useGridSection.ts

```diff
@@ -0,0 +1,109 @@
+import { useCallback, useEffect, useState } from "react";
+import { useASTGuidelinesOptions } from "../../../hooks/useASTGuidelinesOptions";
+import {
+    AntibioticSection,
+    QuestionnaireSection,
+} from "../../../../domain/entities/Questionnaire/QuestionnaireSection";
+import {
+    Question,
+    isASTResultsQuestion,
+    isAddNewAntibioticQuestion,
+    isAntibioticQuestion,
+    isAntibioticValueQuestion,
+    isSpeciesQuestion,
+} from "../../../../domain/entities/Questionnaire/QuestionnaireQuestion";
+import { QuestionnaireStage } from "../../../../domain/entities/Questionnaire/Questionnaire";
+import _c from "../../../../domain/entities/generic/Collection";
+
+export const useGridSection = (
+    speciesSection: QuestionnaireSection,
+    antibioticStage: QuestionnaireStage,
+    updateQuestion: (question: Question) => void
+) => {
+    const { getAntibioticOptions } = useASTGuidelinesOptions();
+    const [gridOptions, setGridOptions] = useState<string[]>();
+    const [antibioticSets, setAntibioticSets] = useState<AntibioticSection[]>();
+
+    useEffect(() => {
+        const speciesQuestion = speciesSection.questions.find(isSpeciesQuestion);
+        if (speciesQuestion) {
+            const options = getAntibioticOptions(speciesQuestion);
+            setGridOptions(options);
+        }
+
+        const antibioticSections = antibioticStage.sections.filter(
+            section => section.isAntibioticSection
+        );
+        const antibioticGroups: AntibioticSection[] = _c(
+            antibioticSections.map(section => {
+                const antibioticQuestion = section?.questions.find(isAntibioticQuestion);
+                const astQuestion = section?.questions.find(isASTResultsQuestion);
+                const valueQuestion = section?.questions.find(isAntibioticValueQuestion);
+                const addNewAntibioticQuestion = section?.questions.find(
+                    isAddNewAntibioticQuestion
+                );
+
+                if (
+                    !antibioticQuestion ||
+                    !astQuestion ||
+                    !valueQuestion ||
+                    !addNewAntibioticQuestion
+                )
+                    return null;
+
+                const antiBioticSet: AntibioticSection = {
+                    antibioticQuestion: antibioticQuestion,
+                    astResultsQuestion: astQuestion,
+                    valueQuestion: valueQuestion,
+                    addNewAntibioticQuestion: addNewAntibioticQuestion,
+                };
+
+                return antiBioticSet;
+            })
+        )
+            .compact()
+            .value();
+
+        setAntibioticSets(antibioticGroups);
+    }, [antibioticStage.sections, getAntibioticOptions, gridOptions, speciesSection.questions]);
+
+    const updateSpeciesQuestion = useCallback(
```

Comment:
I don't know the context but, this is ok? `if isSpeciesQuestion` is going to updateQuestion everytime `if`s inside are fulfilled and also at the end after the `if isSpeciesQuestion`, the question is going to be updated again just with question? Maybe it's better to create a question object to be updated and at the end pass this object to the function updateQuestion?

Reply:
Hi @anagperal ,yes this is correct. Whenever a species question is updated, we need to reset all the associated antibiotic questions . So, the last updateQuestion updates the species question itself and the rest of the updatequestion calls are for the antiobiotic questions which are of 4 types :)